code/game/objects/structures/viewscreen.dm

```diff
@@ -151,8 +170,16 @@
 
 /obj/structure/viewscreen_controller/Topic(href,href_list)
 	..()
-	if(href_list["switch"])
-		linked.view_mode = !linked.view_mode
+	if(!usr.canUseTopic(src))
+		return
+
+	if(href_list["refreshscreens"])
+		linked_viewscreens = get_area_viewscreens()
+
+	if(href_list["switchmode"])
+		var/obj/structure/viewscreen/V = locate(href_list["switchmode"])
```

Comment:
instead of re locating the screen you should be passing the index of the screen in linked_viewscreens